### PR TITLE
Windows fix

### PR DESCRIPTION
**Pull request by ghost-luck**
Include files fix affect MSVC only.

Static initialization order fix affects both MSVC and MinGW GCC. Before that change there were some runtime errors at the library initialization stage:
https://isocpp.org/wiki/faq/ctors#static-init-order

### CODE REVIEW

File: src/tiramisu_expr.cpp [hunk 1]
```diff
@@ -3,6 +3,11 @@
 
 namespace tiramisu
 {
+bool global::auto_data_mapping = false;
```
Comment (rbaghdadi) [inline src/tiramisu_expr.cpp]: Why did you move them here? I assume because the original order created a problem.

Would the code work if you move them to the header file core.h instead? These variables do not belong semantically to the file expr.h since they are general and not related to expressions only. So moving them somewhere else makes more sense.

Comment (rbaghdadi) [reply src/tiramisu_expr.cpp]: .

**Comment by ghost-luck** [comment]
> So moving them somewhere else makes more sense.

Yes, we can move them here.

I made additional testing of the library under memory debugger (Dr. Memory) and found some bugs which were leading to Visual Studio problems. They were all related to my ISL build.

Now both MinGW and Visual Studio build of Tiramisu works fine.

File: src/tiramisu_expr.cpp (hunk 1 above)
Comment (ghost-luck) [reply src/tiramisu_expr.cpp]: > So moving them somewhere else makes more sense.

Yes, we can move them here.

I made additional testing of the library under memory debugger (Dr. Memory) and found some bugs which were leading to Visual Studio problems. They were all related to my ISL build.

Now both MinGW and Visual Studio build of Tiramisu works fine.
